spec/api/keys/keys_spec.rb

```diff
@@ -95,207 +63,146 @@
-iKJOAhHcJcowWjHChP6DSbwXWgnPJa0dsUuBA26Laplw+5NcQ/4WWcKxkidG1nQM
-NfsEUbJLynvnNoAIAqfC1LU4hDaHQBUobF/shucxGFvugW+cH3uhGPUNlyEWGtcj
-RgpQ9222VMRaSNndAaMDKQKBgAQSx+0GEEobGosXmz6k2UjHQ3QwQW16aWQIia3x
-f/Ttz8lSwjVeHPca3pc4P2miN6ZSRDUOrhA7lEWiKH0vrjlPh6i9tuG9Ph3nNnuc
-eA5QMFm93kJERfGTQz4hyKDJWaaiXZQyG63cAxrZcBBGVqB6fxT3WaQAvKYaQpSV
-6SJ9AoGBAL7Zg4LPhHtQ5xwOeLgqKMXJJvX56F8BiYMXpNqVhvrSwgYOCUXiouDu
-0GFNwlun+qgTAxYaJWS55c3jUe8ati/OrMOr62WlfUzNoY1ziJfbL/S0xXKC/SBE
-slFU7wOOLWfOPho31gOm1siG675SXTg8efNm/koyU0LK4KuT8WJE
------END RSA PRIVATE KEY-----"
-
-  $alt_private_key_filepath = "/tmp/alt-private-key-#{Time.now.to_i}"
-
-  $alt_public_key = "-----BEGIN PUBLIC KEY-----
-MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA2YewTVJI52IrzCwv5KzX
-vufn6QQkqiwEdh3OPdaeolfciywZZFlRB57fJbK/cLuqM4/lxe/9Xxoa42Ct2RZo
-V5i2EYSNSa2snQwupRSge2UatR5Z+QcLaCXqCR/n+6c5+JSSGwOKOuuiD4EKAGf2
-RwE6wlba9rWArmnMr/sGA5Ox6ogpoE+jmD1FkKALDMJ7LZaDZ2kJVXXYKKnijnZH
-O++LhAu3gC547wnfO4JoX2StYqHN9OjTJK9YRw45Zq7l3W54W+6dHTLzsgHravWe
-d6t3m7JfBIZft6U4ZtPY5LGK1qPSPDGu8bwn6vJ62Tcf3fzuI2i6x3WWiM9XQilO
-uQIDAQAB
------END PUBLIC KEY-----"
-
-  $alt_public_key_filepath = "/tmp/alt-public-key-#{Time.now.to_i}.pub"
-
-  # TODO remove this after we have APIs in place, since we can just pass
-  # strings then
-  #
-  # write our key files to temp
+  let(:org_base_url) { "#{platform.server}/organizations/#{$org['name']}" }
+  let(:new_user_list_keys_response) do
+    [
+      { "name" => "default", "uri" => "#{platform.server}/users/#{user['name']}/keys/default", "expired" => false}
+    ]
+  end
+  let(:new_client_list_keys_response) do
+    [
+      { "name" => "default", "uri" => "#{org_base_url}/clients/#{client['name']}/keys/default", "expired" => false }
+    ]
+  end
+
+  def delete_client_key(org, client, keyname)
+    system("chef-server-ctl delete-client-key #{org} #{client} #{keyname}")
+  end
+  def delete_user_key(user, keyname)
+    system("chef-server-ctl delete-user-key #{user} #{keyname}")
+  end
+
+  def add_client_key(org, client, key, options = {})
+    # POST { public_key => keys[key][:public] }
+    # name => options[:key_name]
+    command = "chef-server-ctl add-client-key #{org} #{client} #{keys[key][:path]}"
+    command << " --key-name #{options[:key_name]}" unless options[:key_name].nil?
+    command << " --expiration-date #{options[:expires]}" unless options[:expires].nil?
+    system(command)
+  end
+  def add_user_key(user, key, options = {})
+    command = "chef-server-ctl add-user-key #{user} #{keys[key][:path]}"
+    command << " --key-name #{options[:key_name]}" unless options[:key_name].nil?
+    command << " --expiration-date #{options[:expires]}" unless options[:expires].nil?
+    system(command)
+  end
+  def list_user_keys(user, requestor)
+    get("#{platform.server}/users/#{user}/keys", requestor)
+  end
+  def list_client_keys(org, client, requestor)
+    get("#{platform.server}/organizations/#{org}/clients/#{client}/keys", requestor)
+  end
```

Comment:
I'm not really a fan of the amount of helper functions. Spec tests should be very explicit. Just my opinion.